### ISSUE TITLE
Bug: PriorityClass is not set for mining processes

### ISSUE THREAD

**Issue by boro2000** [MultiPoolMiner/MultiPoolMiner#1299]
Since the mining executables are still running with the wrong priority, I did some digging into the code of function Start-SubProcess. 

The Start-Job starts another pwsh-process - inside this, the mining process is started with the "&" command. Regarding that construct, the thing is: `(Get-Process -Id $PID)` inside the Start-SubProcess function, returns the process id of the pwsh process and not the started mining executable. Therefor the PriorityClass of the pwsh-process is changed and not that of the mining executable. I verified this by having the process id prompted and checking, which process it was.

For NiceHash's excavator, it is different: it is started via the APIs/Nicehash.ps1, there, the PriorityClass is not set all.

**Comment by aaronsace** [comment]
I can't check it at the moment but sub-processes should inherit the priority: 
`Start-Job {(Get-Process -Id $PID).PriorityClass = "BelowNormal"; & notepad | Write-Output}`

Excavator improvements have been delayed until v2.7.3 as to get the release out faster.

**Comment by boro2000** [comment]
I just checked with the "notepad"-example + PriorityClass = "AboveNormal".
No, it doesn't inherit the priority. Notepad is launched with "Normal".

**Comment by UselessGuru** [comment]
$Proc = "(Get-Process -Id `$PID).PriorityClass = 'AboveNormal'; & notepad | Write-Output"
Start-Job ([ScriptBlock]::Create($Proc))

I suspect the "`" (backtick) in '`$PID' to be the culprit.

**Comment by UselessGuru** [comment]
> No, that backtick is ok, in this case. 

Try removing it.

The trouble with jobs is that you do not see any error message the trow.

`(Get-Process -Id `$PID).PriorityClass` will always return 'normal' because there is no process with id `$PID

**Comment by boro2000** [comment]
The backtick does it's job:

```
$Proc = "(Get-Process -Id `$PID).PriorityClass = 'High'; `$PID; (Get-Process -Id `$PID).PriorityClass; & notepad | Write-Output"
$Job = Start-Job ([ScriptBlock]::Create($Proc))
Get-Job -Id $Job.Id | Receive-Job
```
.. results into ..
```
75032

RunspaceId                       Value
----------                       -----
6633d6e1-...                      High 
```

.. where 75032 is the actual process ID of the subprocess started by Start-Job. Looking at the real priority of the subprocess (with Process Lasso, for example) it shows the right priority "High".
Anyway: the notepad get's "Normal".

.. strange!

**Comment by aaronsace** [comment]
Bit lost with this but I've just tested on my laptop and MultiPoolMiner set the priority. 

@boro2000 Were you having issues or was you just asking? 

![image](https://user-images.githubusercontent.com/21113644/36613180-a10cb736-18d0-11e8-84e1-0fa7911347c1.png)

**Comment by boro2000** [comment]
Yes, I am having this issue.
I dicovered this, because on mining setups with a weak CPU, the MPM script influences the hashrate of the mining executables, when MPM is in it's recalculating phase. Changing the processpriority to something above normal, helps a lot, then.

Therefor I decided to give the miners more process priority.
In the Miner-class method "StartMining" there is this line:

`            $this.Process = Start-SubProcess -FilePath $this.Path -ArgumentList $this.Arguments -LogPath $Global:ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath(".\Logs\$($this.Name)-$($this.Port)_$(Get-Date -Format "yyyy-MM-dd_HH-mm-ss").txt") -WorkingDirectory (Split-Path $this.Path) -Priority ($this.Type | ForEach-Object {if ($_ -eq "CPU") {-2}else {-1}} | Measure-Object -Maximum | Select-Object -ExpandProperty Maximum)
`

.. there, I changed -1 to 1:
`{if ($_ -eq "CPU") {-2}else {1}} |`

.. and I expected to get AboveNormal PriorityClass for the miner executables. But it seems to be limited to "Normal".
